src/System.Security.Cryptography.Encryption.Aes/src/Internal/Cryptography/AesImplementation.cs

```diff
@@ -41,11 +41,13 @@ private ICryptoTransform CreateTransform(byte[] rgbKey, byte[] rgbIV, bool encry
         {
             if (rgbKey == null)
                 throw new ArgumentNullException("key");
-            int keySize = rgbKey.Length * BitsPerByte;
-            if (!keySize.IsLegalSize(this.LegalKeySizes))
-                throw new ArgumentException(SR.Format(SR.Cryptography_InvalidKeySize, "key"));
+
+            long keySize = rgbKey.Length * (long)BitsPerByte;
+            if (keySize > int.MaxValue || !((int)keySize).IsLegalSize(this.LegalKeySizes))
+                throw new ArgumentException(SR.Cryptography_InvalidKeySize, "key");
+
             if (rgbIV != null && rgbIV.Length * BitsPerByte != BlockSize)
-                throw new ArgumentException(SR.Format(SR.Cryptography_InvalidIVSize, "iv"));
+                throw new ArgumentException(SR.Cryptography_InvalidIVSize, "iv");
```

Comment:
I just noticed this multiplication could overflow as well.  Will address in subsequent commit.